### PR TITLE
Hw6 exceptions

### PR DESCRIPTION
**Pull request by badyornyi**
Added test for rescuing exception

### CODE REVIEW

File: main_project/our_module.rb [hunk 1]
```diff
@@ -291,4 +299,40 @@ def issue_watched?
...
     @driver.find_element(:xpath => "//a[.='#{issue_subject}']").click
   end
+
+  def create_project_with_name
+    create_project_open_page
+    create_project_fill_form
+    create_project_commit
+  end
+
+  def open_project(project_name)
```
Comment (d-konovalov) [inline main_project/our_module.rb]: too big method
split it into logical parts please

**Review by d-konovalov — CHANGES_REQUESTED** [review]

File: main_project/our_module.rb [hunk 2]
```diff
@@ -306,10 +306,7 @@ def create_project_with_name
     create_project_commit
   end
 
-  def open_project(project_name)
-    @wait.until {@driver.find_element(:css => '#projects-index').displayed?}
-    projects = @driver.find_elements(:css => '.project.root.leaf')
-
+  def is_project_in_list?(project_name, projects)
```
Comment (d-konovalov) [inline main_project/our_module.rb]: could you please explain what does this method should do, please?

Comment (badyornyi) [reply main_project/our_module.rb]: "is_project_in_list?" method looks for project name (project that has to be opened) in the all existing projects names list.
If there is no such project in the list - it sets @exist to false

Comment (d-konovalov) [reply main_project/our_module.rb]: I would suggest you to change it

1. get all projects titles
2. verify that this array includes project name you're looking for

**Review by d-konovalov — CHANGES_REQUESTED** [review]

File: main_project/our_module.rb [hunk 3]
```diff
@@ -306,19 +306,21 @@ def create_project_with_name
...
-  def is_project_in_list?(project_name, projects)
-    projects.map do |project|
-      if project.text == project_name
-        @exist = true
-        break
-      else
-        @exist = false
-      end
+  def get_project_titles
+    @driver.find_elements(:css => '.project.root.leaf').map! do |project_element|
```
Comment (d-konovalov) [inline main_project/our_module.rb]: here is some syntax sugar:
map block in this case can be replaced with this:


```ruby
ary.map(&:text)
```

File: main_project/our_module.rb [hunk 4]
```diff
@@ -306,19 +306,21 @@ def create_project_with_name
...
+      project_element.text
     end
   end
 
+  def is_project_in_list?(project_name)
```
Comment (d-konovalov) [inline main_project/our_module.rb]: here is a more suitable method for such cases:


```ruby
ary.include?(element)
```

**Review by d-konovalov — APPROVED** [review]

**Pull request merged**